pkg/lwip/contrib/sock/lwip_sock.c

```diff
@@ -151,6 +151,23 @@ static bool _addr_on_netif(int family, int netif_num, const ip_addr_t *addr)
     p = (ep)->port;
 #endif
 
+static void _convert_ip_addr(ip_addr_t *lwip_addr, int family,
+                             const void *sock_addr, size_t sock_addr_size)
+{
+    memcpy(lwip_addr, sock_addr, sock_addr_size);
+#if LWIP_IPV6 && LWIP_IPV4
+    if (family == AF_INET6) {
+        ip6_addr_clear_zone(&lwip_addr->u_addr.ip6);
+    }
+    lwip_addr->type = lwip_af_to_ip_addr_type(family);
+#elif LWIP_IPV6
+    (void)family;
```

Comment:
Nope. That's the frustrating thing with lwIP. Everything is `#ifdef`'d. So if there is only one type of address (or rather one type of IP implementation) compiled in, family doesn't need to be pulled in.

Reply:
I suspected something like that, thanks for clarifying

Reply:
See [lwIP's `ip_addr.h`](http://git.savannah.nongnu.org/cgit/lwip.git/tree/src/include/lwip/ip_addr.h) if you want to get a head-ache ;-).

Reply:
(but to boil it down: `ip_addr_t` is only a distinct type, when both IPv4 and IPv6 are in, otherwise it's just an alias for the distinctive `ipx_addr_t`.

Reply:
Good to know that I'm not the only one having some issues with that file.